src/Statement.php

```diff
@@ -154,19 +153,11 @@ public function bindParam($param, &$variable, int $type = ParameterType::STRING,
     /**
      * {@inheritDoc}
      *
-     * @deprecated Statement::execute() is deprecated, use Statement::executeQuery() or executeStatement() instead
-     *
      * @throws Exception
      */
-    public function execute(?array $params = null): Result
+    private function execute(array $params = []): Result
     {
-        Deprecation::triggerIfCalledFromOutside(
-            'doctrine/dbal',
-            'https://github.com/doctrine/dbal/pull/4580',
-            'Statement::execute() is deprecated, use Statement::executeQuery() or Statement::executeStatement() instead'
-        );
-
-        if ($params !== null) {
+        if ($params !== []) {
             $this->params = $params;
         }
```

Comment:
```suggestion
        $this->params = $params;
```

Assigning an empty array should be safe here, as it has always been.

Reply:
I believe the idea is that if you're calling `execute()` without parameters, it doesn't override the parameters bound via `bindParam()` or `bindValue()`.

Reply:
I see, thanks.